packages/eas-cli/src/build/createContext.ts

```diff
@@ -80,7 +80,7 @@ export async function createBuildContextAsync<T extends Platform>({
   };
   Analytics.logEvent(BuildEvent.BUILD_COMMAND, trackingCtx);
 
-  const commonContext: CommonContext<T> = {
+  const commonContext = {
```

Comment:
I'm not convinced about this change. With this line here and `as ...` below we lose type safety.

Reply:
This is a really good point. I did some digging and I think this PR could go quite a bit deeper. Will investigate soon, but I actually think I may abandon this effort for now (this goes a lot deeper).